mineinabyss-features/src/main/kotlin/com/mineinabyss/relics/StarCompassListener.kt

```diff
@@ -0,0 +1,51 @@
+package com.mineinabyss.relics
+
+import com.mineinabyss.components.playerData
+import com.mineinabyss.components.relics.StarCompass
+import com.mineinabyss.deeperworld.world.section.section
+import com.mineinabyss.geary.autoscan.AutoScan
+import com.mineinabyss.geary.ecs.accessors.TargetScope
+import com.mineinabyss.geary.ecs.accessors.building.get
+import com.mineinabyss.geary.ecs.api.systems.TickingSystem
+import com.mineinabyss.helpers.bossbarCompass
+import com.mineinabyss.idofront.items.editItemMeta
+import com.mineinabyss.looty.tracking.toGearyOrNull
+import net.kyori.adventure.bossbar.BossBar
+import net.kyori.adventure.text.Component
+import org.bukkit.Location
+import org.bukkit.Material
+import org.bukkit.entity.Player
+import org.bukkit.inventory.meta.CompassMeta
+import kotlin.time.Duration.Companion.seconds
+
+@AutoScan
+class StarCompassSystem : TickingSystem(interval = 0.1.seconds) {
+    private val TargetScope.player by get<Player>()
+    private val compassBar = BossBar.bossBar(Component.text(":arrow_null:"), 1.0f, BossBar.Color.WHITE, BossBar.Overlay.PROGRESS)
+
+    override fun TargetScope.tick() {
+        val compass = player.inventory.contents.firstOrNull {
+            if (it == null) return@firstOrNull false
+            it.toGearyOrNull(player)?.has<StarCompass>() ?: return@firstOrNull false
+        } ?: return
+        val starCompass = compass.toGearyOrNull(player)?.get<StarCompass>() ?: return
+
+        val sectionCenter = player.location.section?.region?.center ?: return
+
+        starCompass.compassLocation = Location(player.world, sectionCenter.x.toDouble(), 0.0, sectionCenter.z.toDouble())
+
+        // Let player toggle between having a bossbar-compass or item compass
+        if (player.playerData.starCompassToggle) {
```

Comment:
Really like this concept btw, however i'd prefer it be done by adding or removing a component from the item. I think the reason set persisting wasn't working for you is because of player instanced item (I assume?) which I don't remember making a mechanism for so and I'm working on that code rn so might as well update that.